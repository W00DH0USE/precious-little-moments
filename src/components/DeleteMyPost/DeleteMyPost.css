.deletePost {
    top: 10rem;
}

#deletePostTitle {
    color: white;
}