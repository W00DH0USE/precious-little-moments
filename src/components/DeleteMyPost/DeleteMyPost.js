import React from 'react';
import './DeleteMyPost.css';

function DeleteMyPost(props) {
    const post = props.posts.filter(post => post.id === props.postId)
    return (
        <>
        <div className='modal-wrapper deletePost'
                style={{
                    transform: props.show ? 'translateY(0vh)' : 'translateY(-100vh)',
                    opacity: props.show ? '1' : '0'
                }}>
                <div className='modal-header'>
                    <h3 id='deletePostTitle'>Delete Post "{post[0].post_title}"?</h3>
                </div>
                <div className='modal-body'>
                    <p>Post Deletions Cannot be Undone!</p>
                </div>
                <div className="modal-footer">
                    <button className="btn-cancel" onClick={props.close}><span>CLOSE</span></button>
                    <button className="btn-continue" type='button' onClick={props.handleDelete}><span>CONTINUE</span></button>
                </div>
            </div>
        </>
    )
}

export default DeleteMyPost;