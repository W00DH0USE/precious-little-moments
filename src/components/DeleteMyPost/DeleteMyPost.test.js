import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DeleteMyPost from './DeleteMyPost';


describe(`DeleteMyPost component`, () => {
    const post = [
        {
            post_title: 'I might relapse!',
            post_content: 'I am having a terrible day! Nothing is going right and I feel like I am going to relapse.',
            owner: 1
        },
        {
            post_title: 'Tips!',
            post_content: 'I have been sober for 3 years. What has helped me and my journey is support and keeping busy!',
            owner: 1
        },
        {
            post_title: 'Feeling Good!',
            post_content: 'I am loving quitting my addiction. I feel so much better!',
            owner: 1
        },
    ]
    it('renders without crashing', () => {
        const wrapper = shallow(<DeleteMyPost posts={post} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});