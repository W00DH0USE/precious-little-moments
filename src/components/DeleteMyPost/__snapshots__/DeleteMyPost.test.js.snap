// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMyPost component renders without crashing 1`] = `
<Fragment>
  <div
    className="modal-wrapper"
    style={
      Object {
        "opacity": "0",
        "transform": "translateY(-100vh)",
      }
    }
  >
    <div
      className="modal-header"
    >
      <h3>
        Delete Post 
        I might relapse!
        ?
      </h3>
    </div>
    <div
      className="modal-body"
    >
      <p>
        Post Deletions Cannot be Undone!
      </p>
    </div>
    <div
      className="modal-footer"
    >
      <button
        className="btn-cancel"
      >
        CLOSE
      </button>
      <button
        className="btn-continue"
        type="button"
      >
        CONTINUE
      </button>
    </div>
  </div>
</Fragment>
`;
